### DISCUSSION TITLE
passing in a parsing context didn't work for precedence! and cache_left_rec

### DISCUSSION THREAD

**Discussion by lfnoise** [kevinmehall/rust-peg#312] [Q&A]
I created a working parser for a non-trivial language syntax in a few hours, so that was pretty neat, thanks. 

I needed to pass a &mut symbol table struct down the tree of parsing rules. This worked fine for all but precedence climbing and left recursive rules. The left recursion gave me an error message that arguments weren't supported.. fair enough. Precedence climbing just wouldn't compile. It gave an error about not being able to capture a mutable in the rule closure and needing a FnMut instead of a Fn. Left recursion was easy to solve, but creating separate nested rules for a dozen levels of precedence by hand was not so nice. My parser works, but is not as simple it would be with the precedence! macro. 

Did I do something wrong? Is there a way to pass an argument down into the @ (next lower precedence rule) somehow? 
[Here's the code.](https://gist.github.com/lfnoise/cd666681203df2cc53548c9659fd9b96)
Thank you.

**Comment by kevinmehall** [comment]
Looks like a bug. Opened https://github.com/kevinmehall/rust-peg/issues/313. I'm not sure there's an easy fix though, because there are two closures involved in the implementation of `precedence!{}`, so they can't both mutably reborrow the reference.

**Comment by kevinmehall** [comment]
As a workaround, if you avoid the `&mut` by doing something like `&RefCell<SymbolTable>`, I think it will work.
